py/desisim/templates.py

```diff
@@ -704,11 +708,11 @@ def make_galaxy_templates(self, nmodel=100, zrange=(0.6, 1.6), magrange=(21.0, 2
                     colormask = np.repeat(1, nbasechunk)
                 else:
                     colormask = self.colorcuts_function(
-                        gflux=synthnano[:, 1],
-                        rflux=synthnano[:, 2],
-                        zflux=synthnano[:, 4],
-                        w1flux=synthnano[:, 6],
-                        w2flux=synthnano[:, 7])
+                        gflux=synthnano[:, 0],
+                        rflux=synthnano[:, 1],
+                        zflux=synthnano[:, 2],
+                        w1flux=synthnano[:, 3],
+                        w2flux=synthnano[:, 4])
```

Comment:
`synthnano` as a 2D array predates this PR, but it is somewhat fragile to have these magic indices tied to the order of the filters specified in the constructor `self.decamwise = filters.load_filters...`.  Especially since part of the motivation for unpacking the `decam_flux` 2D arrays is to get away from these magic indices, could you do the same here with code something like:
```python
synthnano = dict()
for key in enumerate(maggies.columns):
    synthnano[key] = 1E9 * maggies[key] * magnorm # nanomaggies
...
colormask = self.colorcuts_function(
    gflux=synthnano['decam2014-g'],
    rflux=synthnano['decam2014-r'],
    zflux=synthnano['decam2014-z'],
    w1flux=synthnano['wise2010-W1'],
    w2flux=synthnano['wise2010-W2])
```
Same comment applies to several other sections with magic indices into `synthnano` arrays. This caught my eye because the the mapping of indices to filters changed; using a dictionary keyed by filter name would avoid that problem.